modules/pre-commit.nix

```diff
@@ -351,7 +361,11 @@ in
                 echo 1>&2 "    2. remove .pre-commit-config.yaml"
                 echo 1>&2 "    3. add .pre-commit-config.yaml to .gitignore"
               else
-                ln -fs ${configFile} "''${GIT_WC}/.pre-commit-config.yaml"
+                if ${boolToString cfg.addGcRoot}; then
+                  nix-store --add-root "''${GIT_WC}/.pre-commit-config.yaml" --indirect --realise ${configFile}
```

Comment:
Why doesn't it symlink in this case?

Reply:
The `nix-store` command creates the symlink. The result is the same as `ln -s` + registering the GC root with Nix.